hooks/tests/.claude/hooks/check_root_files.sh

```
@@ -0,0 +1,107 @@
+        if [[ "$FILENAME" == "$allowed" ]]; then
+            IS_ALLOWED=true
+            break
+        fi
+    done
+
+    if [ "$IS_ALLOWED" = false ]; then
+        # Block the operation
+        cat <<EOF
+{
+  "decision": "block",
+  "reason": "${RED}🚨 ROOT FILE POLLUTION DETECTED${NC}\\n\\nFile: $FILENAME\\nLocation: Project root\\n\\n${YELLOW}Files should be organized in proper directories:${NC}\\n• Documentation → docs/\\n• Tests → tests/ or testing_*/\\n• Scripts → scripts/ or claude_command_scripts/\\n• Configs → configs/ or .config/\\n• Hooks → hooks/\\n• Examples → examples/\\n• Temporary → tmp/ or temp/\\n\\n${GREEN}Allowed root files:${NC} $(printf '%s, ' \"${ALLOWED_ROOT_FILES[@]}\" | sed 's/, $//')\\n\\nPlease move this file to an appropriate subdirectory.",
```

Shellcheck (0.10.0)

[error] Double quote array expansions to avoid re-splitting elements.

(SC2068)